src/index.js

```diff
@@ -44,74 +55,72 @@ const create = (privateKey, value, seq, lifetime, callback) => {
  * @param {string} value value to be stored in the record.
  * @param {number} seq number representing the current version of the record.
  * @param {string} expiration expiration datetime for record in the [RFC3339]{@link https://www.ietf.org/rfc/rfc3339.txt} with nanoseconds precision.
- * @param {function(Error, entry)} [callback]
+ * @returns {Promise<IpnsEntry>} entry
  */
-const createWithExpiration = (privateKey, value, seq, expiration, callback) => {
+const createWithExpiration = (privateKey, value, seq, expiration) => {
   const validityType = ipnsEntryProto.ValidityType.EOL
-  _create(privateKey, value, seq, expiration, validityType, callback)
+  return _create(privateKey, value, seq, expiration, validityType)
 }
 
-const _create = (privateKey, value, seq, isoValidity, validityType, callback) => {
-  sign(privateKey, value, validityType, isoValidity, (error, signature) => {
-    if (error) {
-      log.error('record signature creation failed')
-      return callback(Object.assign(new Error('record signature verification failed'), { code: ERRORS.ERR_SIGNATURE_CREATION }))
-    }
+const _create = async (privateKey, value, seq, isoValidity, validityType) => {
+  const signature = await sign(privateKey, value, validityType, isoValidity)
 
-    const entry = {
-      value: value,
-      signature: signature,
-      validityType: validityType,
-      validity: isoValidity,
-      sequence: seq
-    }
+  const entry = {
+    value: value,
+    signature: signature,
+    validityType: validityType,
+    validity: isoValidity,
+    sequence: seq
+  }
 
-    log(`ipns entry for ${value} created`)
-    return callback(null, entry)
-  })
+  log(`ipns entry for ${value} created`)
+  return entry
 }
 
 /**
  * Validates the given ipns entry against the given public key.
  *
  * @param {Object} publicKey public key for validating the record.
- * @param {Object} entry ipns entry record.
- * @param {function(Error)} [callback]
+ * @param {IpnsEntry} entry ipns entry record.
+ * @returns {Promise}
  */
-const validate = (publicKey, entry, callback) => {
+const validate = async (publicKey, entry) => {
   const { value, validityType, validity } = entry
   const dataForSignature = ipnsEntryDataForSig(value, validityType, validity)
 
   // Validate Signature
-  publicKey.verify(dataForSignature, entry.signature, (err, isValid) => {
-    if (err || !isValid) {
-      log.error('record signature verification failed')
-      return callback(Object.assign(new Error('record signature verification failed'), { code: ERRORS.ERR_SIGNATURE_VERIFICATION }))
+  let isValid
+  try {
+    isValid = await publicKey.verify(dataForSignature, entry.signature)
+  } catch (err) {
```

Comment:
```suggestion
  } catch () {
```

Reply:
It turns out this is actually a syntax error in JS - the `catch()` must have a variable inside it, eg `catch(e)`

Reply:
do not agree with that linting rule, but let's keep it then!